core/node/consensus/src/tests.rs

```diff
@@ -661,6 +662,62 @@ async fn test_centralized_fetcher(from_snapshot: bool, version: ProtocolVersionI
     .unwrap();
 }
 
+#[test_casing(4, Product((FROM_SNAPSHOT,VERSIONS)))]
+#[tokio::test]
+async fn test_simulated_l1_status_api(from_snapshot: bool, version: ProtocolVersionId) {
+    zksync_concurrency::testonly::abort_on_panic();
+    let ctx = &ctx::test_root(&ctx::RealClock);
+    let rng = &mut ctx.rng();
+
+    scope::run!(ctx, |ctx, s| async {
+        let validator_pool = ConnectionPool::test(from_snapshot, version).await;
+        let (mut validator, runner) =
+            testonly::StateKeeper::new(ctx, validator_pool.clone()).await?;
+        s.spawn_bg(runner.run(ctx).instrument(tracing::info_span!("validator")));
+
+        // API server needs at least 1 L1 batch to start.
+        validator.seal_batch().await;
+        let api = validator.connect(ctx).await?;
+
+        // If the main node has no L1 batch certificates,
+        // the first one to sign should be `last_sealed_batch + 1`.
+        let status = api.fetch_simulated_l1_status().await?;
+        assert_eq!(
+            status.next_batch_to_commit,
+            validator.last_sealed_batch() + 1
+        );
+
+        // Insert a cert, then check again.
+        validator.push_random_block(rng).await;
+        validator.seal_batch().await;
```

Comment:
It would be good to check here again after the L1 batch is inserted, but before the batch QC is attached to the previous one, that the API returns what you expect, which at the moment IIUC is the original batch, not affected by the new one. 

I am not sure that's correct though, as I suspect if the main node was restarted it would move on to the last batch, so it might as well return the last until it has a certificate, or the one from the snapshot, if it wants to fill gaps.

Reply:
Yeah, exactly, so we agree, right? To clarify, what I wanted to say is that adding the test would enshrine the behaviour whether the last one is returned, or the first one, because currently AFAICS it is immediately followed up with a cert, which makes the fact that `first_` exists unobservable in the test.

Reply:
I've added the assert